core/src/main/java/org/elasticsearch/search/searchafter/SearchAfterBuilder.java

```diff
@@ -47,13 +45,34 @@
 /**
  *
  */
-public class SearchAfterBuilder implements ToXContent, FromXContentBuilder<SearchAfterBuilder>, Writeable<SearchAfterBuilder> {
-    public static final SearchAfterBuilder PROTOTYPE = new SearchAfterBuilder();
+public class SearchAfterBuilder implements ToXContent, Writeable<SearchAfterBuilder> {
     public static final ParseField SEARCH_AFTER = new ParseField("search_after");
     private static final Object[] EMPTY_SORT_VALUES = new Object[0];
 
     private Object[] sortValues = EMPTY_SORT_VALUES;
 
+    public SearchAfterBuilder() {
+    }
+
+    /**
+     * Read from a stream.
+     */
+    public SearchAfterBuilder(StreamInput in) throws IOException {
+        int size = in.readVInt();
+        sortValues = new Object[size];
+        for (int i = 0; i < size; i++) {
+            sortValues[i] = in.readGenericValue();
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVInt(sortValues.length);
+        for (Object fieldValue : sortValues) {
+            out.writeGenericValue(fieldValue);
```

Comment:
I had the same thought, but I guess the point is to properly validate things when they get set to the builder, so that not supported values will never be serialized. I assume that we do that already, otherwise we should.